core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -842,6 +862,32 @@ object Parser extends ParserInstances {
     }
   }
 
+  /** If this right parser fails to parse its input with an epsilon error,
+    * try the left parser instead.
+    *
+    * If the right parser fails with an arresting error, the left parser
+    * won't be tried.
+    *
+    * Backtracking may be used on the left parser to allow the right
+    * one to pick up after any error, resetting any state that was
+    * modified by the left parser.
+    */
+  def eitherOr[A, B](right: Parser[B], left: Parser[A]): Parser[Either[A, B]] =
+    oneOf(right.map(Right(_)) :: left.map(Left(_)) :: Nil)
+
+  /** If this right parser fails to parse its input with an epsilon error,
+    * try the left parser instead.
+    *
+    * If the right parser fails with an arresting error, the left parser
+    * won't be tried.
+    *
+    * Backtracking may be used on the left parser to allow the right
+    * one to pick up after any error, resetting any state that was
+    * modified by the left parser.
+    */
+  def eitherOr1[A, B](right: Parser1[B], left: Parser1[A]): Parser1[Either[A, B]] =
```

Comment:
What about saying first and second to avoid the confusion with names? The types will help us not get confused with the order of the args?